ENIGMAsystem/SHELL/Graphics_Systems/General/GSfont.cpp

```diff
@@ -1287,28 +1261,34 @@ void draw_text_ext_color(gs_scalar x, gs_scalar y,variant vstr,gs_scalar sep, gs
   }
 }
 
-unsigned font_get_texture(int fnt) {
-  get_font_null(f,fnt,-1);
-  return f.texture;
+unsigned font_get_texture(int id) {
+  const SpriteFont& fnt = sprite_fonts.get(id);
+  return fnt.texture;
 }
 
-unsigned font_get_texture_width(int fnt) {
-  get_font_null(f,fnt,-1);
-  return f.twid;
+unsigned font_get_texture_width(int id) {
+  const SpriteFont& fnt = sprite_fonts.get(id);
+  return fnt.twid;
 }
 
-unsigned font_get_texture_height(int fnt) {
-  get_font_null(f,fnt,-1);
-  return f.thgt;
+unsigned font_get_texture_height(int id) {
+  const SpriteFont& fnt = sprite_fonts.get(id);
+  return fnt.thgt;
 }
 
-unsigned font_height(int fnt) {
-  get_font_null(f,fnt,-1);
-  return f.height;
+unsigned font_height(int id) {
+  const SpriteFont& fnt = sprite_fonts.get(id);
+  return fnt.height;
 }
 
 void draw_set_font(int fnt) {
-  enigma::currentfont = fnt;
+  if (fnt == -1 || sprite_fonts.exists(fnt))
+    enigma::currentfont = fnt;
+  #ifdef DEBUG_MODE
+  else
+    DEBUG_MESSAGE("Requested font asset " + std::to_string(fnt) + " does not exist.", MESSAGE_TYPE::M_USER_ERROR);
+  #endif
```

Comment:
GM8.1 and GMSv1.4 will not show this for font -1, -2, or any other non-existing font. However, I am ok with this change because the manual only says that -1 is acceptable for the default font.